webapp/src/redux/ducks/studyMaps.ts

```diff
@@ -622,7 +621,23 @@ export default createReducer(initialState, (builder) => {
     })
     .addCase(deleteStudyMapNodeTemp, (draftState, action) => {
       const { studyId, nodeId } = action.payload;
-      delete draftState.entities[studyId]?.nodes[nodeId];
+      const entity = draftState.entities[studyId];
+      if (entity) {
+        delete entity.nodes[nodeId];
+      }
+    })
+    .addCase(deleteStudyMapNode.fulfilled, (draftState, action) => {
+      const { studyId, nodeId } = action.payload;
+      const entity = draftState.entities[studyId];
+      if (entity) {
```

Comment:
Why so you check the entity here?

Reply:
not the same action `deleteStudyMapNode.fulfilled`